tools/auth/vk.go

```diff
@@ -0,0 +1,82 @@
+package auth
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/vk"
+)
+
+var _ Provider = (*Vk)(nil)
+
+// NameVk is the unique name of the Vk provider.
+const NameVk string = "vk"
+
+// Vk allows authentication via Vk OAuth2.
+type Vk struct {
+	*baseProvider
+}
+
+// NewVkProvider creates new Vk provider instance with some defaults.
+//
+// Docs: https://dev.vk.com/api/oauth-parameters
+func NewVkProvider() *Vk {
+	return &Vk{&baseProvider{
+		scopes:     []string{"email"},
+		authUrl:    vk.Endpoint.AuthURL,
+		tokenUrl:   vk.Endpoint.TokenURL,
+		userApiUrl: "https://api.vk.com/method/users.get?fields=photo_max,screen_name&v=5.131",
```

Comment:
Wouldn't `users.get` returns all users associated to the current VK account? How we decide which one to choose (currently it seems that always the first one will be selected)?

I'm having difficulties navigating through the VK russian docs, but wouldn't [`account.getProfileInfo`](https://dev.vk.com/method/account.getProfileInfo) be more suitable for retrieving the user data?

Reply:
[`account.getProfileInfo`](https://dev.vk.com/method/account.getProfileInfo) works only for token that was recieved through Implicit Flow, and can't be used on server.

> Этот метод можно вызвать с ключом доступа пользователя, полученным в Standalone-приложении через [Implicit Flow](https://dev.vk.com/api/access-token/implicit-flow-user).

`users.get` indeed returns a current user if `user_ids` is not set.